### PR TITLE
Small refactor : EventStream as a dataclass

### PR DESCRIPTION
**Pull request by tofarr**
**Convert eventstream to a dataclass**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
This shortens the code, and adds useful `__eq__` / `__repr__` methods

### CODE REVIEW

**Comment by xingyaoww** [comment]
Maybe unrelated -- but we have `BaseModel` from pydantic and `dataclass`. Do you think if we can refactor this to pydantic BaseModel instead of dataclass so we also get the benefit for type validation?

File: openhands/events/stream.py [hunk 1]
```diff
@@ -29,24 +30,17 @@ def session_exists(sid: str, file_store: FileStore) -> bool:
...
-    _lock: threading.Lock
-
-    def __init__(self, sid: str, file_store: FileStore):
-        self.sid = sid
-        self.file_store = file_store
-        self._subscribers = {}
-        self._cur_id = 0
-        self._lock = threading.Lock()
-        self._reinitialize_from_file_store()
+    _subscribers: Dict[str, list[Callable]] = field(default_factory=dict)
```
Comment (enyst) [inline openhands/events/stream.py:39]: s/Dict/dict

**Review by enyst — APPROVED** [review]

File: openhands/events/stream.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import asyncio
 import threading
+from dataclasses import dataclass, field
 from datetime import datetime
 from enum import Enum
-from typing import Callable, Iterable
+from typing import Callable, Dict, Iterable
```
Comment (xingyaoww) [inline openhands/events/stream.py:6]:
```suggestion
from typing import Iterable
```

File: openhands/events/stream.py (hunk 1 above)
Comment (xingyaoww) [inline openhands/events/stream.py:39]:
```suggestion
    _subscribers: dict[str, list[callable]] = field(default_factory=dict)
```

**Review by xingyaoww — APPROVED** [review]

File: openhands/events/stream.py (hunk 1 above)
Comment (tofarr) [reply openhands/events/stream.py]: Unfortunately ruff complains about the use of `callable` as a type hint, so I can't do this one for now

**Pull request merged**